oracle_fdw.c

```diff
@@ -732,6 +734,69 @@ _PG_init(void)
 	on_proc_exit(&exitHook, PointerGetDatum(NULL));
 }
 
+/*
+ * Deparse LIMIT clause translated for ORACLE into FETCH FIRST N ROWS ONLY
+ * If OFFSET is set then do not append clause at all, the limit +offset
+ * will be applied localy.
+ */
+char *
+deparseLimit(PlannerInfo *root, struct OracleFdwState *fdwState, RelOptInfo *baserel)
+{
+	StringInfoData limit_clause;
+
+	initStringInfo(&limit_clause);
+
+	if (root->parse->limitCount != NULL && root->parse->limitOffset == NULL)
+	{
+		if (IsA(root->parse->limitCount, Const) && !((Const *) root->parse->limitCount)->constisnull)
+		{
+			int limit_at_end = false;
+
+			/*
+			 * we must be sure that there is no ORDER BY clause that
+			 * appears after the LIMIT clause otherwise we will have
+			 * wrong results.
+			 */
```

Comment:
I still don't understand this.
How can an `ORDER BY` be after a `LIMIT`?
Perhaps the code is right, but the comment should explain that.

Reply:
There is a typo here I mean GROUP BY not ORDER BY and actually as GROUP BY are not pushed this code can be removed. Will do.

Reply:
Hum, no I remember now. Actually I have picked-up the code from a DB2 FDW I'm developing but I have not taking care of the current orafce_architecture and Oracle only support  a FETCH FIRST clause at end of the query. It can not be placed before and ORDER BY or a GROUP BY clause, I will remove this useless code.